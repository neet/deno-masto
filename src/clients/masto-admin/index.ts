export * from './masto-admin';
export * from './params';
