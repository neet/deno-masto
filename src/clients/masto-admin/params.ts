export interface AdminFetchAccountParams {
  /** Filter for local accounts? */
  local?: boolean | null;
  /** Filter for remote accounts? */
  remote?: boolean | null;
  /** Filter by the given domain */
  byDomain?: string | null;
  /** Filter for currently active accounts? */
  active?: boolean | null;
  /** Filter for currently pending accounts? */
  pending?: boolean | null;
  /** Filter for currently disabled accounts? */
  disabled?: boolean | null;
  /** Filter for currently silenced accounts? */
  silenced?: boolean | null;
  /** Filter for currently suspended accounts? */
  suspended?: boolean | null;
  /** Username to search for */
  username?: string | null;
  /** Display name to search for */
  displayName?: string | null;
  /** Lookup a user with this email */
  email?: string | null;
  /** Lookup users by this IP address */
  ip?: string | null;
  /** Filter for staff accounts? */
  staff?: boolean | null;
}

export interface AdminFetchReportsParams {
  resolved?: boolean | null;
  accountId?: string | null;
  targetAccountId?: string | null;
  byTargetDomain?: string | null;
}

export type AccountActionType = 'none' | 'disable' | 'silence' | 'suspend';

export interface AdminActionAccountParams {
  /** Type of action to be taken. Enumerable oneOf: `none` `disable` `silence` `suspend` */
  type?: AccountActionType;
  /** ID of an associated report that caused this action to be taken */
  reportId?: string;
  /** ID of a preset warning */
  warningPresetId?: string | null;
  /** Additional text for clarification of why this action was taken */
  text?: string | null;
  /** Whether an email should be sent to the user with the above information. */
  sendEmailNotification?: boolean | null;
}
