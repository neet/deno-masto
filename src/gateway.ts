export * from "./gateway/flatten-data.ts";
export * from "./gateway/decorators.ts";
export * from "./gateway/gateway-impl.ts";
export * from "./gateway/gateway.ts";
