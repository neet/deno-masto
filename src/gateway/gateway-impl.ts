import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
import { camelCase, snakeCase } from 'change-case';
import normalizeUrl from 'normalize-url'; // eslint-disable-line import/default
import querystring, { ParsedUrlQueryInput } from 'querystring';
import semver from 'semver';

import { Instance } from '../entities';
import {
  MastoConflictError,
  MastoForbiddenError,
  MastoGoneError,
  MastoNotFoundError,
  MastoRateLimitError,
  MastoUnauthorizedError,
  MastoUnprocessableEntityError,
} from '../errors';
import { createFormData } from './create-form-data';
import { EventHandlerImpl } from './event-handler-impl';
import {
  Gateway,
  GatewayConstructorParams,
  LoginParams,
  PaginateNext,
} from './gateway';
import { isAxiosError } from './is-axios-error';
import { transformKeys } from './transform-keys';

/**
 * Mastodon network request wrapper
 * @param params Optional params
 */
export class GatewayImpl implements Gateway<AxiosRequestConfig> {
  /** Configured axios instance */
  private axios: AxiosInstance;
  /** URI of the instance */
  private _uri = '';
  /** Streaming API URL of the instance */
  private _streamingApiUrl = '';
  /** Version of the current instance */
  version: string;
  /** API token of the user */
  accessToken?: string;

  /**
   * @param params Parameters
   */
  constructor(params: GatewayConstructorParams<AxiosRequestConfig>) {
    this.uri = params.uri;
    this.version = params.version;

    if (params.accessToken) {
      this.accessToken = params.accessToken;
    }

    if (params.streamingApiUrl) {
      this.streamingApiUrl = params.streamingApiUrl;
    }

    this.axios = axios.create({
      baseURL: this.uri,
      transformResponse: this.transformResponse,
      ...(params.defaultOptions ?? {}),
    });

    this.axios.interceptors.request.use(this.transformConfig);
    this.axios.defaults.headers.common['Content-Type'] = 'application/json';
    this.axios.defaults.headers.common.Authorization = `Bearer ${this.accessToken}`;
  }

  get uri() {
    return this._uri;
  }

  set uri(uri: string) {
    this._uri = normalizeUrl(uri);
  }

  get streamingApiUrl() {
    return this._streamingApiUrl;
  }

  set streamingApiUrl(streamingApiUrl: string) {
    this._streamingApiUrl = normalizeUrl(streamingApiUrl);
  }

  /**
   * Login to Mastodon
   * @param params Parameters
   * @return Instance of Mastodon class
   */
  static async login<T extends typeof GatewayImpl>(
    this: T,
    _params: LoginParams,
  ) {
    const params = { ..._params, version: '0.0.0' };
    const gateway = new this(params) as InstanceType<T>;
    const instance = await gateway.get<Instance>('/api/v1/instance');

    gateway.version = instance.version;
    gateway.streamingApiUrl = instance.urls.streamingApi;

    return gateway;
  }

  /**
   * Transform JSON to JS object
   * @param response Response object
   * @return Parsed entity
   */
  private transformResponse(data: string, _headers: unknown) {
    try {
      return transformKeys(JSON.parse(data), camelCase);
    } catch {
      return data;
    }
  }

  /**
   * Encode data in request options and add authorization / content-type header
   * @param config Axios config
   * @return New config
   */
  private transformConfig({
    params,
    data,
    ...originalConfig
  }: AxiosRequestConfig) {
    const config: AxiosRequestConfig = { ...originalConfig };

    if (params) {
      config.params = transformKeys(params, snakeCase);
    }

    if (data) {
      config.data = transformKeys(data, snakeCase);
    }

    switch (config.headers['Content-Type']) {
      case 'application/json':
        config.data = JSON.stringify(config.data);

        return config;

      case 'multipart/form-data':
        config.data = createFormData(config.data);

        // In Node.js, axios doesn't set boundary data to the header
        // so set it manually by using getHeaders of form-data node.js package
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if (typeof (config.data as any).getHeaders === 'function') {
          config.headers = {
            ...config.headers,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            ...(config.data as any).getHeaders(),
          };
        }

        return config;

      default:
        return config;
    }
  }

  /**
   * Wrapper function for Axios
   * @param options Axios options
   * @param parse Whether parse response before return
   * @return Parsed response object
   */
  protected async request<T>(options: AxiosRequestConfig) {
    try {
      return await this.axios.request<T>(options);
    } catch (error) {
      if (!isAxiosError(error)) {
        throw error;
      }

      const status = error?.response?.status;
      const message =
        error?.response?.data?.error ?? 'Unexpected error occurred';

      switch (status) {
        case 401:
          throw new MastoUnauthorizedError(message);
        case 403:
          throw new MastoForbiddenError(message);
        case 404:
          throw new MastoNotFoundError(message);
        case 409:
          throw new MastoConflictError(message);
        case 410:
          throw new MastoGoneError(message);
        case 422:
          throw new MastoUnprocessableEntityError(message);
        case 429:
          throw new MastoRateLimitError(message);
        default:
          throw error;
      }
    }
  }

  /**
   * HTTP GET
   * @param url URL to request
   * @param params Query strings
   * @param options Fetch API options
   * @param parse Whether parse response before return
   */
  async get<T>(
    path: string,
    params: unknown = {},
    options?: AxiosRequestConfig,
  ) {
    const response = await this.request<T>({
      method: 'GET',
      url: path,
      params,
      ...options,
    });

    return response.data;
  }

  /**
   * HTTP POST
   * @param url URL to request
   * @param data Payload
   * @param options Fetch API options
   * @param parse Whether parse response before return
   */
  async post<T>(
    path: string,
    data: unknown = {},
    options?: AxiosRequestConfig,
  ) {
    const response = await this.request<T>({
      method: 'POST',
      url: path,
      data,
      ...options,
    });

    return response.data;
  }

  /**
   * HTTP PUT
   * @param path Path to request
   * @param data Payload
   * @param options Fetch API options
   * @param parse Whether parse response before return
   */
  async put<T>(path: string, data: unknown = {}, options?: AxiosRequestConfig) {
    const response = await this.request<T>({
      method: 'PUT',
      url: path,
      data,
      ...options,
    });

    return response.data;
  }

  /**
   * HTTP DELETE
   * @param path Path to request
   * @param data jPayload
   * @param options Fetch API options
   * @param parse Whether parse response before return
   */
  async delete<T>(
    path: string,
    data: unknown = {},
    options?: AxiosRequestConfig,
  ) {
    const response = await this.request<T>({
      method: 'DELETE',
      url: path,
      data,
      ...options,
    });

    return response.data;
  }

  /**
   * HTTP PATCH
   * @param path Path to request
   * @param data Payload
   * @param options Fetch API options
   * @param parse Whether parse response before return
   */
  async patch<T>(
    path: string,
    data: unknown = {},
    options?: AxiosRequestConfig,
  ) {
    const response = await this.request<T>({
      method: 'PATCH',
      url: path,
      data,
      ...options,
    });

    return response.data;
  }

  /**
   * Connect to a streaming
   * @param path Path to stream
   * @param params Query parameters
   * @return Instance of EventEmitter
   */
  stream(path: string, params: ParsedUrlQueryInput = {}) {
    const protocols = [];

    // Since v2.8.4, it is supported to pass access token with`Sec-Websocket-Protocol`
    // https://github.com/tootsuite/mastodon/pull/10818
    if (this.accessToken && this.version && semver.gte(this.version, '2.8.4')) {
      protocols.push(this.accessToken);
    } else if (this.accessToken) {
      params.accessToken = this.accessToken;
    }

    const encodedParams = querystring.stringify(
      transformKeys<ParsedUrlQueryInput>(params, snakeCase),
    );

    const url =
      this.streamingApiUrl +
      path +
      (Object.keys(params).length ? '?' + encodedParams : '');

    return new EventHandlerImpl().connect(url, protocols);
  }

  /**
   * Generate an iterable of the pagination.
   * @param initialURL Path for the endpoint
   * @param initialParams Query parameter
   * @return Async iterable iterator of the pages.
   * See also [MDN article about generator/iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators)
   */
  async *paginate<T, U>(
    initialUrl: string,
    initialParams?: U,
  ): AsyncGenerator<T, void, PaginateNext<U> | undefined> {
    let nextUrl: string | undefined = initialUrl;
    let nextParams = initialParams;

    while (nextUrl) {
      const response: AxiosResponse<T> = await this.request<T>({
        method: 'GET',
        url: nextUrl,
        params: nextParams,
      });

      // Yield will be argument of next()
      const options = yield response.data;
      // Get next URL from "next" in the link header
      const linkHeaderNext = response.headers?.link?.match(
        /<(.+?)>; rel="next"/,
      )?.[1];

      if (options?.reset) {
        nextUrl = initialUrl;
        nextParams = initialParams;
      } else {
        nextUrl = options?.url ?? linkHeaderNext;
        nextParams = options?.params;
      }
    }
  }
}
