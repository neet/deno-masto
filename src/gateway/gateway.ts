import { ParsedUrlQueryInput } from 'querystring';

import { EventHandler } from './event-handler';

/**
 * Argument of `Gateway.paginate().next()`.
 * When reset = true specified, other params won't be accepted
 */
export type PaginateNext<Params> =
  | {
      reset: boolean;
      url?: undefined;
      params?: undefined;
    }
  | {
      reset?: undefined;
      url?: string;
      params?: Params;
    };

export interface Gateway<Options = undefined> {
  uri: string;
  streamingApiUrl: string;
  version?: string;
  accessToken?: string;
  get<T>(path: string, params: unknown, options?: Options): Promise<T>;
  post<T>(path: string, data: unknown, options?: Options): Promise<T>;
  put<T>(path: string, data: unknown, options?: Options): Promise<T>;
  patch<T>(path: string, data: unknown, options?: Options): Promise<T>;
  delete<T>(path: string, data: unknown, options?: Options): Promise<T>;
  stream(path: string, params: ParsedUrlQueryInput): Promise<EventHandler>;
  paginate<T, U>(
    url: string,
    params?: U,
  ): AsyncGenerator<T, void, PaginateNext<U>>;
}

export interface GatewayConstructorParams<T = undefined> {
  /** URI of the instance */
  uri: string;
  /** Streaming API URL */
  streamingApiUrl?: string;
  /** Version of the instance */
  version: string;
  /** Access token of the user */
  accessToken?: string;
  /** Axios configurations. See [Axios'](https://github.com/axios/axios#request-config) docs */
  defaultOptions?: T;
}

export type LoginParams = Omit<
  GatewayConstructorParams,
  'version' | 'streamingApiUrl'
>;

export interface GatewayConstructor<Options = undefined> {
  new (params: GatewayConstructorParams<Options>): Gateway;
  login(params: LoginParams): Promise<Gateway>;
}
