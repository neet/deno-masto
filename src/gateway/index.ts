export * from './create-form-data';
export * from './decorators';
export * from './event-handler-impl';
export * from './event-handler';
export * from './gateway-impl';
export * from './gateway';
export * from './is-axios-error';
