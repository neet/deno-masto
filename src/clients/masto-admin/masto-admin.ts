import { AdminAccount, AdminReport } from "../../entities.ts";
import { available, GatewayImpl } from "../../gateway.ts";
import {
  AdminActionAccountParams,
  AdminFetchAccountParams,
  AdminFetchReportsParams,
} from "./params.ts";

/** Mastodon Moderation API client */
export class MastoAdmin extends GatewayImpl {
  /**
   * View accounts matching certain criteria for filtering, up to 100 at a time.
   * Pagination may be done with the HTTP Link header in the response.
   * @param params Parameters
   * @return Array of AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  fetchAccounts(params?: AdminFetchAccountParams) {
    return this.get<AdminAccount[]>("/api/v1/admin/accounts", params);
  }

  /**
   * View admin-level information about the given account.
   * @param id ID of the account
   * @return AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  fetchAccount(id: string) {
    return this.get<AdminAccount>(`/api/v1/admin/accounts/${id}`);
  }

  /**
   * Perform an action against an account and log this action in the moderation history.
   * @param id g ID of the account
   * @param params Params
   * @return Account
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  actionAccount(id: string, params: AdminActionAccountParams) {
    return this.post<AdminAccount>(
      `/api/v1/admin/accounts/${id}/action`,
      params,
    );
  }

  /**
   * Approve the given local account if it is currently pending approval.
   * @param id ID of the account
   * @return AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  approveAccount(id: string) {
    return this.post<AdminAccount>(`/api/v1/admin/accounts/${id}/approve`);
  }

  /**
   * Reject the given local account if it is currently pending approval.
   * @param id ID of the account
   * @return AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  rejectAccount(id: string) {
    return this.post<AdminAccount>(`/api/v1/admin/accounts/${id}/reject`);
  }

  /**
   * Re-enable a local account whose login is currently disabled.
   * @param id ID of the account
   * @return AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  enableAccount(id: string) {
    return this.post<AdminAccount>(`/api/v1/admin/accounts/${id}/enable`);
  }

  /**
   * Unsilence a currently silenced account.
   * @param id ID of the account
   * @return AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  unsilenceAccount(id: string) {
    return this.post<AdminAccount>(`/api/v1/admin/accounts/${id}/unsilence`);
  }

  /**
   * Unsuspend a currently suspended account.
   * @param id ID of the account
   * @return AdminAccount
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  unsuspendAccount(id: string) {
    return this.post<AdminAccount>(`/api/v1/admin/accounts/${id}/unsuspend`);
  }

  /**
   * View all reports. Pagination may be done with HTTP Link header in the response.
   * @param params Parameters
   * @return Array of AdminReport
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  fetchReports(params?: AdminFetchReportsParams) {
    return this.get<AdminReport[]>("/api/v1/admin/reports", params);
  }

  /**
   * View information about the report with the given ID.
   * @param id ID of the report
   * @return AdminReport
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  fetchReport(id: string) {
    return this.get<AdminReport>(`/api/v1/admin/reports/${id}`);
  }

  /**
   * Claim the handling of this report to yourself.
   * @param id ID of the report
   * @return AdminReport
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  assignReportToSelf(id: string) {
    return this.post<AdminReport>(`/api/v1/admin/reports/${id}/assign_to_self`);
  }

  /**
   * Unassign a report so that someone else can claim it.
   * @param id ID of the report
   * @return AdminReport
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  unassignReport(id: string) {
    return this.post<AdminReport>(`/api/v1/admin/reports/${id}/unassign`);
  }

  /**
   * Mark a report as resolved with no further action taken.
   * @param id ID of the report
   * @return AdminReport
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  resolveReport(id: string) {
    return this.post<AdminReport>(`/api/v1/admin/reports/${id}/resolve`);
  }

  /**
   * Reopen a currently closed report.
   * @param id ID of the report
   * @return AdminReport
   * @see https://docs.joinmastodon.org/methods/admin/
   */
  @available({ since: "2.9.1" })
  reopenReport(id: string) {
    return this.post<AdminReport>(`/api/v1/admin/reports/${id}/reopen`);
  }
}
