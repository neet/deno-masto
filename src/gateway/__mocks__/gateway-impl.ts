import { Gateway } from '../gateway';

export const mockGet = jest.fn();
export const mockPost = jest.fn();
export const mockDelete = jest.fn();
export const mockPatch = jest.fn();
export const mockPut = jest.fn();
export const mockStream = jest.fn();
export const mockPaginate = jest.fn();

export class GatewayImpl implements Gateway {
  uri = 'https://example.com';
  version = '99.99.9';
  streamingApiUrl = 'wss://example.com/stream';
  get = mockGet;
  post = mockPost;
  delete = mockDelete;
  patch = mockPatch;
  put = mockPut;
  stream = mockStream;
  paginate = mockPaginate;
}
