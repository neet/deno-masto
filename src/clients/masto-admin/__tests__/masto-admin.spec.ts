// @ts-ignore
// prettier-ignore
import { Gateway, mockDelete, mockGet, mockPaginate, mockPatch, mockPost, mockPut, mockStream } from '../../../gateway/gateway-impl';
import { MastoAdmin } from '../masto-admin';

jest.mock('../../../gateway/gateway-impl');

describe('MastoAdmin', () => {
  const masto = new MastoAdmin({
    uri: 'https://example.com',
    version: '99.99.9',
    streamingApiUrl: 'wss://example.com/stream',
  });

  beforeEach(async () => {
    masto.version = '99.99.9'; // avoid version error
    mockGet.mockClear();
    mockPost.mockClear();
    mockDelete.mockClear();
    mockPut.mockClear();
    mockPatch.mockClear();
    mockPaginate.mockClear();
    mockStream.mockClear();
  });

  test('fetchAccounts', async () => {
    await masto.fetchAccounts();
    expect(mockGet).toBeCalledTimes(1);
    expect(mockGet).toMatchSnapshot();
  });

  test('fetchAccount', async () => {
    await masto.fetchAccount('123');
    expect(mockGet).toBeCalledTimes(1);
    expect(mockGet).toMatchSnapshot();
  });

  test('enableAccount', async () => {
    await masto.enableAccount('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('approveAccount', async () => {
    await masto.approveAccount('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('rejectAccount', async () => {
    await masto.rejectAccount('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('unsilenceAccount', async () => {
    await masto.unsilenceAccount('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('unsuspendAccount', async () => {
    await masto.unsuspendAccount('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('fetchReports', async () => {
    await masto.fetchReports();
    expect(mockGet).toBeCalledTimes(1);
    expect(mockGet).toMatchSnapshot();
  });

  test('fetchReport', async () => {
    await masto.fetchReport('123');
    expect(mockGet).toBeCalledTimes(1);
    expect(mockGet).toMatchSnapshot();
  });

  test('assignReportToSelf', async () => {
    await masto.assignReportToSelf('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('unassignReport', async () => {
    await masto.unassignReport('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('reopenReport', async () => {
    await masto.reopenReport('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('resolveReport', async () => {
    await masto.resolveReport('123');
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });

  test('actionAccount', async () => {
    await masto.actionAccount('123', {
      type: 'disable',
      reportId: '123123',
      warningPresetId: '123123',
      text: 'Your account have been disabled',
      sendEmailNotification: false,
    });
    expect(mockPost).toBeCalledTimes(1);
    expect(mockPost).toMatchSnapshot();
  });
});
