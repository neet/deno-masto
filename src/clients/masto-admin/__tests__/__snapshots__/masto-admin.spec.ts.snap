// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MastoAdmin actionAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123/action",
      Object {
        "reportId": "123123",
        "sendEmailNotification": false,
        "text": "Your account have been disabled",
        "type": "disable",
        "warningPresetId": "123123",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin approveAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123/approve",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin assignReportToSelf 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/reports/123/assign_to_self",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin enableAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123/enable",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin fetchAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin fetchAccounts 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin fetchReport 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/reports/123",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin fetchReports 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/reports",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin rejectAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123/reject",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin reopenReport 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/reports/123/reopen",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin resolveReport 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/reports/123/resolve",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin unassignReport 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/reports/123/unassign",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin unsilenceAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123/unsilence",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`MastoAdmin unsuspendAccount 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/api/v1/admin/accounts/123/unsuspend",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
